### PR TITLE
Feature/user signup signin

### PR DESCRIPTION
**Pull request by purisaurabh**
This PR includes : 
      1) User SignUp Page
       2) User SignIn Page

### CODE REVIEW

File: app/src/components/UserForm.tsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
+"use client";
+
+interface UserFormProps {
```
Comment (sourabh-josh) [inline app/src/components/UserForm.tsx:3]: please move this below module imports

File: app/src/components/UserForm.tsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  formType: "signup" | "signin";
+}
+
+import { useState } from "react";
+import "../app/globals.css";
+import Link from "next/link";
+import { Button, buttonVariants } from "./ui/button";
+import { cn } from "../lib/utils";
+
+interface UserFormProps {
```
Comment (sourabh-josh) [inline app/src/components/UserForm.tsx:13]: why define the same props twice please remove this

File: app/src/components/UserForm.tsx [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+  formType: "signup" | "signin";
+}
+
+const UserForm = (props: UserFormProps) => {
+  const { formType } = props;
+
+  const isSignup = formType === "signup";
+
+  const [formData, setFormData] = useState({
```
Comment (sourabh-josh) [inline app/src/components/UserForm.tsx:22]: please use `react-hook-form` and `zod` packages to handle form related tasks 
refer to [lingo.ai](https://github.com/joshsoftware/lingo.ai/blob/main/app/src/components/UserForm.tsx)

**Review by sourabh-josh — CHANGES_REQUESTED** [review]

File: app/prettier.config.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  trailingComma: "all",
```
[the remark this answers is not in the document]
Comment (purisaurabh) [reply app/prettier.config.js]: done. I removed config file

File: app/src/components/UserForm.tsx (hunk 1 above)
Comment (purisaurabh) [reply app/src/components/UserForm.tsx]: done

File: app/src/components/UserForm.tsx (hunk 2 above)
Comment (purisaurabh) [reply app/src/components/UserForm.tsx:24]: removed

File: app/src/components/UserForm.tsx [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+  };
+
+  return (
+    <div className="flex flex-col gap-2 w-full justify-center items-center">
+      <form
+        onSubmit={handleSubmit}
+        className="w-full max-w-sm flex flex-col gap-4 justify-center items-center"
+        style={{ fontSize: "14px" }}
```
[the remark this answers is not in the document]
Comment (purisaurabh) [reply app/src/components/UserForm.tsx]: done

File: app/src/components/UserForm.tsx (hunk 3 above)
Comment (purisaurabh) [reply app/src/components/UserForm.tsx]: done

**Pull request merged**